core/input_map.cpp

```diff
@@ -96,19 +98,41 @@ List<StringName> InputMap::get_actions() const {
 	return actions;
 }
 
-List<Ref<InputEvent> >::Element *InputMap::_find_event(List<Ref<InputEvent> > &p_list, const Ref<InputEvent> &p_event, bool p_action_test) const {
+List<Ref<InputEvent> >::Element *InputMap::_find_event(Action p_action, const Ref<InputEvent> &p_event, bool *p_pressed, float *p_strength) const {
 
-	for (List<Ref<InputEvent> >::Element *E = p_list.front(); E; E = E->next()) {
+	for (List<Ref<InputEvent> >::Element *E = p_action.inputs.front(); E; E = E->next()) {
 
 		const Ref<InputEvent> e = E->get();
 
 		//if (e.type != Ref<InputEvent>::KEY && e.device != p_event.device) -- unsure about the KEY comparison, why is this here?
 		//	continue;
 
 		int device = e->get_device();
-		if (device == ALL_DEVICES || device == p_event->get_device())
-			if (e->action_match(p_event))
+		if (device == ALL_DEVICES || device == p_event->get_device()) {
```

Comment:
I would probably just move this to InputEvent::action_match(), and pass deadzone, *pressed and *sterngth as optional arguments there , so we dont harcode it here.

This would add more flexibility for future events, and we could also use deadzone for InputEventJoyButton, which has pressure too on some controllers.

Reply:
Same problem as with event_is_action, since action_match is bound to GDscript I cannot use pointers in the function arguments. I could duplicate the function though, like with event_get_action_status().